drivers/i2c/CMakeLists.txt

```diff
@@ -22,7 +22,11 @@ zephyr_library_sources_ifdef(CONFIG_I2C_EMUL		i2c_emul.c)
 zephyr_library_sources_ifdef(CONFIG_NRFX_TWI		i2c_nrfx_twi.c)
 zephyr_library_sources_ifdef(CONFIG_NRFX_TWIM		i2c_nrfx_twim.c)
 zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWI		i2c_sam_twi.c)
-zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWIHS	i2c_sam_twihs.c)
+if(CONFIG_I2C_RTIO)
+zephyr_library_sources_ifdef(CONFIG_I2C_SAM_TWIHS	i2c_sam_twihs_rtio.c)
```

Comment:
Can this be converted to `(CONFIG_I2C_SAM_TWIHS AND CONFIG_I2C_RTIO)`? Then we can remove the if/else?

Reply:
I tried that at first but cmake wasn’t happy, will try again before marking as ready

Reply:
CMake does not like the boolean expression in the sources_ifdef sadly, so I wrapped the entire rtio subset of sources in an if